include/engine/datafacade/contiguous_internalmem_datafacade.hpp

```diff
@@ -154,7 +155,7 @@ class ContiguousInternalMemoryAlgorithmDataFacade<algorithm::CoreCH>
     : public datafacade::AlgorithmDataFacade<algorithm::CoreCH>
 {
   private:
-    util::ShM<bool, true>::vector m_is_core_node;
+    util::ShM<unsigned, osrm::storage::MemorySetting::SharedMemory>::vector m_is_core_node;
```

Comment:
That helps @danpat - that probably explains why this part was acting up! I really appreciate getting your input on these issues - OSRM is a fantastic piece of software but sometimes slightly complex :)